sources/packages/backend/libs/services/src/notifications/notification/notification-actions.service.ts

```diff
@@ -617,6 +618,67 @@ export class NotificationActionsService {
     );
   }
 
+  /**
+   * Creates e-cert blocked notification for student and ministry.
+   * @param notification notification details.
+   * @param auditUserId user who blocks the ecert.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  async saveEcertBlockedNotification(
+    notification: EcertBlockedNotification,
+    auditUserId: number,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    const studentNotificationTemplateId =
+      await this.notificationMessageService.getTemplateId(
+        NotificationMessageType.StudentNotificationEcertBlocked,
+      );
+    const studentNotificationToSend = {
```

Comment:
Thought of asking you to create a common method for constructing the notificationsToSend. But as we are following this process in the above methods.We can ignore :)
```
async function constructNotificationToSend(
  messageType: NotificationMessageType,
  auditUserId: number,
  notification: EcertBlockedNotification,
  notificationMessageService: NotificationMessageService,
  gcNotifyService: GCNotifyService
): Promise<any> {
  const templateId = await notificationMessageService.getTemplateId(messageType);
  return {
    userId: auditUserId,
    messageType: messageType,
    messagePayload: {
      email_address: messageType === NotificationMessageType.StudentNotificationEcertBlocked ? notification.email : gcNotifyService.ministryToAddress(),
      template_id: templateId,
      personalisation: {
        givenNames: notification.givenNames ?? "",
        lastName: notification.lastName,
        dob: messageType === NotificationMessageType.MinistryNotificationEcertBlocked ? getDateOnlyFormat(notification.dob) : undefined,
        studentEmail: messageType === NotificationMessageType.MinistryNotificationEcertBlocked ? notification.email : undefined,
        applicationNumber: messageType === NotificationMessageType.MinistryNotificationEcertBlocked ? notification.applicationNumber : undefined,
        dateTime: messageType === NotificationMessageType.MinistryNotificationEcertBlocked ? getDateTimeOnPSTTimeZone() : undefined,
      },
    },
  };
}
```